File: api/FSE_ProjMgr/FSE_ProjMgr/Models/Task.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;

namespace FSE_ProjMgr.Models
{
    public class Task
    {
        public int TaskId { get; set; }
        public int? ParentID { get; set; }
        public int? ProjectID { get; set; }
        public string TaskName { get; set; }
        public DateTime? StartDate { get; set; }
        public DateTime? EndDate { get; set; }
        public int? Priority { get; set; }
        public int Status { get; set; }
        public User User { get; set; }
        public string ParentTaskName { get; set; }
    }
}

File: ui/src/app/models/task.ts
import { User } from "./user";

export class Task {
    public StartDate: string;
    public EndDate: string;
    public TaskName: string;
    public ProjectID: number;
    public TaskId: number;
    public priority: number;
    public ParentID: number;
    public Status: number;
    public user: User;
    public parentTaskName: string;

    constructor() {
        this.user = new User();
    }
}
